### ISSUE TITLE
CCR  failed: org.apache.lucene.index.CorruptIndexException: checksum failed (hardware problem?) 

### ISSUE THREAD

**Issue by bydeath** [elastic/elasticsearch#73549]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.9.0、7.9.2、7.9.3
It's ok on 7.8.1 and 7.10.0 and 7.11.1, 7.12.1, 7.13.0.

**Plugins installed**: []

**JVM version** (`java -version`):Elasticsearch builtin JVM

**OS version** (`uname -a` if on a Unix-like system): Ubuntu 20.04 

**Description of the problem including expected versus actual behavior**:
CCR following failed an indexing completly index, logs is below

> rg.apache.lucene.index.CorruptIndexException: checksum failed (hardware problem?) : expected=1s10c92 actual=6aetaw (resource=name [_17_Lucene80_0.dvd], length [33197279], checksum [1s10c92], writtenBy [8.6.0]) (resource=VerifyingIndexOutput(_17_Lucene80_0.dvd))

I have reproduced the problem on Elasticsearch 7.9.0, 7.9.3 and 7.9.3. While there is no the problem on the verson 7.8.1 and 7.10.0, 7.11.1, 7.12.1, 7.13.0.

**Steps to reproduce**:

Please include a *minimal* but *complete* recreation of the problem,
including (e.g.) index creation, mappings, settings, query etc.  The easier
you make for us to reproduce it, the more likely that somebody will take the
time to look at it.
run the script [github rally/blob/reproduce_ccr_err/recipes/ccr/start.sh](https://github.com/bydeath/rally/blob/reproduce_ccr_err/recipes/ccr/start.sh) 

the script includes these key steps to reproduce the problem 
1. ingest the default benchmark dataset `geonames_copy` into a cluster es1.
2. configure remote cluster in another cluster es2; ccepting trial license on both clusters; wait the ingest is complete and configure follower index for the index `geonames_copy`.
3. check the follower index status, and view the server log of es2.

the response of the command 
```bash
curl -H 'Content-Type: application/json' -XPUT http://localhost:39202/geonames_copy/_ccr/follow\?wait_for_active_shards\=1 -d'{"remote_cluster": "leader","leader_index": "geonames"}'
```
is: 
`{"follow_index_created":true,"follow_index_shards_acked":false,"index_following_started":false}`

the  response of the command
```bash
curl  http://localhost:39202/_cat/indices
```
is

`red open geonames_copy zekuo6mHScSVi-bFnVZgEA 3 0`
**Provide logs (if relevant)**:

> {"type": "server", "timestamp": "2021-05-31T08:26:32,655Z", "level": "WARN", "component": "o.e.c.r.a.AllocationService", "cluster.name": "follower", "node.name": "bff7f1e74c10", "message": "failing shard [failed shard, shard [geonames_copy][0], node[88_LEeI2S8C_0dyc_uOudQ], [P], recovery_source[snapshot recovery [4zkyfBg4R-y1-uGDmpSAIA] from _ccr_leader:_latest_/_latest_], s[INITIALIZING], a[id=jesCWjdPQbmBNLz6f3Vd3Q], unassigned_info[[reason=NEW_INDEX_RESTORED], at[2021-05-31T08:26:30.720Z], delayed=false, details[restore_source[_ccr_leader/_latest_]], allocation_status[fetching_shard_data]], message [failed recovery], failure [RecoveryFailedException[[geonames_copy][0]: Recovery failed on {bff7f1e74c10}{88_LEeI2S8C_0dyc_uOudQ}{E3kKHqrhTP2E-Sr38T9FVQ}{192.168.112.3}{192.168.112.3:9300}{dilmrt}{ml.machine_memory=8158711808, xpack.installed=true, transform.node=true, ml.max_open_jobs=20}]; nested: IndexShardRecoveryException[failed recovery]; nested: IndexShardRestoreFailedException[restore failed]; nested: IndexShardRestoreFailedException[failed to restore snapshot [_latest_/_latest_]]; nested: CorruptIndexException[checksum failed (hardware problem?) : expected=1s10c92 actual=6aetaw (resource=name [_17_Lucene80_0.dvd], length [33197279], checksum [1s10c92], writtenBy [8.6.0]) (resource=VerifyingIndexOutput(_17_Lucene80_0.dvd))]; ], markAsStale [true]]", "cluster.uuid": "7Sw9JabIT7S2luoxlH78xQ", "node.id": "88_LEeI2S8C_0dyc_uOudQ" , 
"stacktrace": ["org.elasticsearch.indices.recovery.RecoveryFailedException: [geonames_copy][0]: Recovery failed on {bff7f1e74c10}{88_LEeI2S8C_0dyc_uOudQ}{E3kKHqrhTP2E-Sr38T9FVQ}{192.168.112.3}{192.168.112.3:9300}{dilmrt}{ml.machine_memory=8158711808, xpack.installed=true, transform.node=true, ml.max_open_jobs=20}",
"at org.elasticsearch.index.shard.IndexShard.lambda$executeRecovery$21(IndexShard.java:2665) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.index.shard.StoreRecovery.lambda$recoveryListener$6(StoreRecovery.java:363) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.index.shard.StoreRecovery.lambda$restore$8(StoreRecovery.java:487) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.xpack.ccr.repository.CcrRepository.lambda$restoreShard$1(CcrRepository.java:324) ~[?:?]",
"at org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:94) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$6.onFailure(ActionListener.java:292) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer.onCompleted(MultiChunkTransfer.java:145) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer.handleItems(MultiChunkTransfer.java:133) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer.access$000(MultiChunkTransfer.java:59) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer$1.write(MultiChunkTransfer.java:78) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.common.util.concurrent.AsyncIOProcessor.processList(AsyncIOProcessor.java:108) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.common.util.concurrent.AsyncIOProcessor.drainAndProcessAndRelease(AsyncIOProcessor.java:96) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.common.util.concurrent.AsyncIOProcessor.put(AsyncIOProcessor.java:84) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer.addItem(MultiChunkTransfer.java:89) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiChunkTransfer.lambda$handleItems$4(MultiChunkTransfer.java:125) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:65) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.action.support.ThreadedActionListener$1.doRun(ThreadedActionListener.java:98) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:710) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]",
"at java.lang.Thread.run(Thread.java:832) [?:?]",
"Caused by: org.elasticsearch.index.shard.IndexShardRecoveryException: failed recovery",
"... 27 more",
"Caused by: org.elasticsearch.index.snapshots.IndexShardRestoreFailedException: restore failed",
"... 25 more",
"Caused by: org.elasticsearch.index.snapshots.IndexShardRestoreFailedException: failed to restore snapshot [_latest_/_latest_]",
"... 23 more",
"Caused by: org.apache.lucene.index.CorruptIndexException: checksum failed (hardware problem?) : expected=1s10c92 actual=6aetaw (resource=name [_17_Lucene80_0.dvd], length [33197279], checksum [1s10c92], writtenBy [8.6.0]) (resource=VerifyingIndexOutput(_17_Lucene80_0.dvd))",
"at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.readAndCompareChecksum(Store.java:1200) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeByte(Store.java:1178) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeBytes(Store.java:1208) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiFileWriter.innerWriteFileChunk(MultiFileWriter.java:126) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiFileWriter.access$000(MultiFileWriter.java:46) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiFileWriter$FileChunkWriter.writeChunk(MultiFileWriter.java:213) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.indices.recovery.MultiFileWriter.writeFileChunk(MultiFileWriter.java:74) ~[elasticsearch-7.9.0.jar:7.9.0]",
"at org.elasticsearch.xpack.ccr.repository.CcrRepository$RestoreSession$1.writeFileChunk(CcrRepository.java:562) ~[?:?]",
"at org.elasticsearch.xpack.ccr.repository.CcrRepository$RestoreSession$1.lambda$executeChunkRequest$1(CcrRepository.java:542) ~[?:?]",
"at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63) ~[elasticsearch-7.9.0.jar:7.9.0]",
"... 6 more"] }

**Comment by original-brownbear** [comment]
Hi @bydeath as you point out this is only an issue on 7.9.x, this was fixed in https://github.com/elastic/elasticsearch/pull/64632 so I'm closing here as there's nothing else to be done. Upgrading to 7.10 or later fixes the issue.

**Comment by bydeath** [comment]
> Hi @bydeath as you point out this is only an issue on 7.9.x, this was fixed in #64632 so I'm closing here as there's nothing else to be done. Upgrading to 7.10 or later fixes the issue.

Thank you!
